test/jdk/java/time/tck/java/time/format/TCKDateTimeParseResolver.java

```diff
@@ -854,32 +854,32 @@ public void test_resolveClockHourOfAmPm(ResolverStyle style, long value, Integer
     @DataProvider(name="resolveAmPm")
     Object[][] data_resolveAmPm() {
         return new Object[][]{
-                {STRICT, 0, 0},
-                {STRICT, 1, 1},
-                {STRICT, -1, null},
-                {STRICT, 2, null},
-
-                {SMART, 0, 0},
-                {SMART, 1, 1},
-                {SMART, -1, null},
-                {SMART, 2, null},
-
-                {LENIENT, 0, 0},
-                {LENIENT, 1, 1},
-                {LENIENT, -1, -1},
-                {LENIENT, 2, 2},
+                {STRICT, 0, LocalTime.of(6, 0), 0},
+                {STRICT, 1, LocalTime.of(18, 0), 1},
```

Comment:
As mentioned in my other comment, this seems odd in STRICT mode.

Reply:
Did you mean in STRICT mode, HOUR_OF_AMPM should default to 0, and to 6 in SMART/LENIENT modes?

Reply:
No. I mean that when resolving AMPM/dayPeriod in strict mode, and there is no HOUR_OF_DAY or HOUR_OF_AMPM, then do not resolve using "half way between"(ie. fail). This will produce a result where `LocalTime` cannot be obtained.

```
var f = DateTimeFormatter.ofPattern("B").withResolverStyle(ResolverStyle.STRICT);
var t = LocalTime.from("at night", f);
```
would throw an exception in STRICT mode (whereas SMART or LENIENT would return a `LocalTime`). Same with pattern "a".

Reply:
Changed to throw an exception in STRICT mode.